lib/preview-web/src/PreviewWeb.integration.test.ts

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import global from 'global';
+import { RenderContext } from '@storybook/store';
+import addons from '@storybook/addons';
+
+import { PreviewWeb } from './PreviewWeb';
+import {
+  componentOneExports,
+  importFn,
+  projectAnnotations,
+  getProjectAnnotations,
+  fetchStoryIndex,
+  emitter,
+  mockChannel,
+  waitForRender,
+} from './PreviewWeb.mockdata';
+
+// PreviewWeb.test mocks out all rendering
+//   - ie. from`renderToDOM()` (stories) or`ReactDOM.render()` (docs) in.
+// This file lets them rip.
+
+jest.mock('@storybook/channel-postmessage', () => () => mockChannel);
+
+jest.mock('./WebView');
+
+const { window, document } = global;
+jest.mock('global', () => ({
+  // @ts-ignore
+  ...jest.requireActual('global'),
+  history: { replaceState: jest.fn() },
+  document: {
+    ...jest.requireActual('global').document,
+    location: {
+      pathname: 'pathname',
+      search: '?id=*',
+    },
+  },
+  FEATURES: {
+    storyStoreV7: true,
```

Comment:
Do we need tests for `storyStoreV7: false`?

Reply:
I'm not sure how heavily to test it. We do have the tests for `start.test.ts` (in this PR: https://github.com/storybookjs/storybook/pull/16006#issuecomment-915997761) which exercises that code path.

They aren't as comprehensive as these tests, but I (hope) the cover the important differences when the flag is off.